lib/ace/mode/rst_highlight_rules.js

```diff
@@ -171,6 +178,66 @@ var RSTHighlightRules = function() {
       }
     ],
 
+    /* Inline code
+     * The parser returns to the "start" state when reaching "``" */
+    "code": [
+      {
+        token : "markup.raw",
+        regex : "(?:\\\\`|[^`])+",
+        next  : "code"
+      },
+      {
+        token : "markup.raw",
+        regex : "``",
+        next  : "start"
+      }
+    ],
+
+    /* Bold (strong) text
+     * The parser returns to the "start" state when reaching "**" */
+    "bold": [
+      {
+        token : "markup.bold",
```

Comment:
It is better to use `defaultToken: "markup.bold",` instead of creating regexp, to exclude  "**"

Something like

```js
{  defaultToken: "string"},
{ token: "string.end", regex: /\*\*/, next: "start" }
```
`/\*\*/` returns to the start, and `defaultToken` assigns class to all unmatched text. 

Btw which code did you use as example, i probably need to update documentation to remove the description of old method.